x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/security/action/apikey/GetApiKeyRequestTests.java

```diff
@@ -120,33 +123,59 @@ public void writeTo(StreamOutput out) throws IOException {
 
-            assertThat(requestFromInputStream.getApiKeyId(), equalTo(getApiKeyRequest.getApiKeyId()));
+            assertThat(requestFromInputStream.getApiKeyId(), equalTo(getApiKeyRequest1.getApiKeyId()));
             // old version so the default for `ownedByAuthenticatedUser` is false
             assertThat(requestFromInputStream.ownedByAuthenticatedUser(), is(false));
         }
         {
+            final GetApiKeyRequest getApiKeyRequest2 = GetApiKeyRequest.builder()
+                .apiKeyName(apiKeyId)
+                .ownedByAuthenticatedUser(true)
+                .withLimitedBy(true)
+                .build();
+            ByteArrayOutputStream outBuffer = new ByteArrayOutputStream();
+            OutputStreamStreamOutput out = new OutputStreamStreamOutput(outBuffer);
+            out.setVersion(randomVersionBetween(random(), Version.V_7_4_0, Version.V_8_4_0));
+            getApiKeyRequest2.writeTo(out);
+
+            InputStreamStreamInput inputStreamStreamInput = new InputStreamStreamInput(new ByteArrayInputStream(outBuffer.toByteArray()));
+            inputStreamStreamInput.setVersion(randomVersionBetween(random(), Version.V_7_4_0, Version.V_8_4_0));
+            GetApiKeyRequest requestFromInputStream = new GetApiKeyRequest(inputStreamStreamInput);
+
+            assertThat(requestFromInputStream.getApiKeyId(), equalTo(getApiKeyRequest2.getApiKeyId()));
+            assertThat(requestFromInputStream.ownedByAuthenticatedUser(), is(true));
+            // old version so the default for `withLimitedBy` is false
+            assertThat(requestFromInputStream.withLimitedBy(), is(false));
+        }
+        {
+            final GetApiKeyRequest getApiKeyRequest3 = GetApiKeyRequest.builder()
+                .apiKeyName(apiKeyId)
+                .withLimitedBy(randomBoolean())
+                .build();
             ByteArrayOutputStream outBuffer = new ByteArrayOutputStream();
             OutputStreamStreamOutput out = new OutputStreamStreamOutput(outBuffer);
-            out.setVersion(randomVersionBetween(random(), Version.V_7_4_0, Version.CURRENT));
-            getApiKeyRequest.writeTo(out);
+            out.setVersion(randomVersionBetween(random(), Version.V_8_5_0, Version.CURRENT));
+            getApiKeyRequest3.writeTo(out);
 
             InputStreamStreamInput inputStreamStreamInput = new InputStreamStreamInput(new ByteArrayInputStream(outBuffer.toByteArray()));
-            inputStreamStreamInput.setVersion(randomVersionBetween(random(), Version.V_7_4_0, Version.CURRENT));
+            inputStreamStreamInput.setVersion(randomVersionBetween(random(), Version.V_8_5_0, Version.CURRENT));
             GetApiKeyRequest requestFromInputStream = new GetApiKeyRequest(inputStreamStreamInput);
 
-            assertThat(requestFromInputStream, equalTo(getApiKeyRequest));
+            assertThat(requestFromInputStream, equalTo(getApiKeyRequest3));
```

Comment:
This change is actually a bug fix for test. It can fail because the new `with_limited_by` field is not available in version before v8.5.0.